ans.js

```diff
@@ -0,0 +1,12 @@
+function getFinalOpenedDoors(n) {
+  // all doors are initially closed
+  const doors = new Array(n).fill(false);
+  for (let i = 1; i <= n; i++) {
+    for (let j = i - 1; j < n; j += i) {
+      doors[j] = !doors[j];
+    }
+  }
+  return doors
+    .map((door, index) => door ? index + 1 : null)
+    .filter(number => number !== null);
+}
```

Comment:
IIFEにして引数は省略をやめましょうか
```suggestion
((number) => {
  // all doors are initially closed
  const doors = new Array(n).fill(false);
  for (let i = 1; i <= n; i++) {
    for (let j = i - 1; j < n; j += i) {
      doors[j] = !doors[j];
    }
  }
  return doors
    .map((door, index) => door ? index + 1 : null)
    .filter(number => number !== null);
})();
```